### PR TITLE
[TVMScript][Fix] Add type hints for more uncovered cases

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to add type hints for some uncovered cases in TVMScript.

cc: @junrushao1994 @vinx13

### CODE REVIEW

**Comment by junrushao1994** [comment]
Thank you so much for acting swiftly on this!! BTW, would you mind also adding T.exp?

**Comment by vinx13** [comment]
@shingjan please address comments above

**Comment by shingjan** [comment]
@vinx13 Comments addressed. I fixed the CI issue by disabling the mypy check on slice index of that two lines.

**Comment by junrushao1994** [comment]
The following two errors should be fixed rather than skipped (if possible)
```
tests/python/unittest/test_tvmscript_type.py: note: In function "different_access_indices":
tests/python/unittest/test_tvmscript_type.py:168:29: error: Slice index must be an integer or None
tests/python/unittest/test_tvmscript_type.py:169:29: error: Slice index must be an integer or None
```

**Comment by vinx13** [comment]
It is an issue of mypy that can't be fixed - it requires slice must be constructed with int rather than custom type
see https://github.com/python/typing/issues/159

File: tests/python/unittest/test_tvmscript_type.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-# pylint: disable=missing-function-docstring,missing-module-docstring,invalid-name,pointless-string-statement
+# pylint: disable=missing-function-docstring,missing-module-docstring,invalid-name,pointless-string-statement,line-too-long
```
Comment (vinx13) [inline tests/python/unittest/test_tvmscript_type.py:17]: rather than disable it, `line-too-long` should be fixed if possible

**Comment by vinx13** [comment]
There are still some CI error. Please make sure it can pass locally

File: tests/python/unittest/test_tvmscript_type.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def different_access_indices(a: T.handle, b: T.handle) -> None:
                     ]
                 )
                 with T.init():
-                    B[vj, vi] = T.exp(B[vi, vj])
+                    B[vj, vi] = T.exp(B[vj, vi], dtype="float32")
```
Comment (junrushao1994) [inline tests/python/unittest/test_tvmscript_type.py:176]: Does it mean that we no longer allow T.exp without dtype explicitly set? If not, what's the default dtype for this?

Comment (junrushao1994) [inline tests/python/unittest/test_tvmscript_type.py:176]: Does it mean that we no longer allow T.exp without dtype explicitly set? If not, what's the default dtype for this?

**Review by junrushao1994 — CHANGES_REQUESTED** [review]

File: tests/python/unittest/test_tvmscript_type.py (hunk 2 above)
Comment (shingjan) [reply tests/python/unittest/test_tvmscript_type.py:176]: If we do need to set a default dtype for intrinsics defined outside of `script/tir/intrinsic`, like `exp` here defined in `tir/op`, is `float32` reasonable here? We could do something like 

```def exp(x: PrimExpr, dtype: str = "float32") -> PrimExpr: ...```

**Comment by junrushao1994** [comment]
Shall we merge this PR?

**Review by junrushao1994 — APPROVED** [review]

**Review by vinx13 — APPROVED** [review]

**Pull request merged**